WordPress/WordPressTest/PeopleServiceTests.swift

```diff
@@ -0,0 +1,224 @@
+                Follower(managedPerson: follower),
+                success: {
+                    done()
+                },
+                failure: {
+                    XCTFail("Unexpected failure: \($0)")
+                    done()
+                }
+            )
+        }
+
+        // Verify the follower is deleted
+        try XCTAssertEqual(mainContext.count(for: request), 0)
+    }
+
+    func testDeleteFollowerFailure() throws {
+        // Load and save followers
+        stub(condition: isPath("/rest/v1.1/sites/\(siteID)/follows")) { _ in
+            HTTPStubsResponse(
+                jsonObject: [
+                    "users":[
+                        [
+                            "ID": 1,
+                            "nice_name": "Nice Name",
+                            "name": "name"
+                        ]
+                    ]
+                ],
+                statusCode: 200,
+                headers: nil
+            )
+        }
+
+        waitUntil { done in
+            self.service.loadFollowersPage(success: { _, _ in done() }, failure: { _ in done() })
+        }
+
+        // Verify if the loaded followers are stored in the database
+        let request = NSFetchRequest<ManagedPerson>(entityName: "Person")
+        request.predicate = NSPredicate(format: "siteID = %@ AND userID = %@", NSNumber(value: siteID), NSNumber(value: 1))
+        try XCTAssertEqual(mainContext.count(for: request), 1)
+        let follower = try XCTUnwrap(mainContext.fetch(request).first)
+        XCTAssertEqual(follower.userID, 1)
+
+        // Make API call to delete the loaded follower
+        stub(condition: isPath("/rest/v1.1/sites/\(siteID)/followers/\(follower.userID)/delete")) { _ in
+            HTTPStubsResponse(jsonObject: [:], statusCode: 500, headers: nil)
+        }
+
+        waitUntil { done in
+            self.service.deleteFollower(
+                Follower(managedPerson: follower),
+                success: {
+                    XCTFail("The failure block should be called instead")
+                    done()
+                },
+                failure: { _ in
+                    done()
+                }
+            )
+        }
+
+        // Verify the follower is not deleted from the database
+        try XCTAssertEqual(mainContext.count(for: request), 1)
+    }
+
+    func testLoadFollowersFailure() {
+        stub(condition: isPath("/rest/v1.1/sites/123/follows")) { _ in
+            HTTPStubsResponse(jsonObject: [:], statusCode: 500, headers: nil)
+        }
+
+        waitUntil { done in
+            self.service.loadFollowersPage(
+                success: { count, _ in
+                    XCTFail("The failure block should be called instead")
+                    done()
+                },
+                failure: { error in
+                    XCTAssertTrue(error is WordPressComRestApiError)
+                    done()
+                }
+            )
+        }
+    }
+
+}
```

Comment:
Imagine the overhead in writing this test without having access to Nimble's `waitUntil { done`... 😅